testutil/retry/retry.go

```diff
@@ -0,0 +1,92 @@
+// Package retry provides support for repeating operations in tests.
+//
+// A sample retry operation looks like this:
+//
+//   func TestX(t *testing.T) {
+//       for r := retry.OneSec(); r.Next(t.FailNow); {
+//           if err := foo(); err != nil {
+//               t.Log("f: ", err)
+//               continue
+//           }
+//           break
+//       }
+//   }
+//
+// A sample retry operation which exits a test with a message
+// looks like this:
+//
+//   func TestX(t *testing.T) {
+//       for r := retry.OneSec(); r.Next(func(){ t.Fatal("foo failed") }); {
+//           if err := foo(); err != nil {
+//               t.Log("f: ", err)
+//               continue
+//           }
+//           break
+//       }
+//   }
+package retry
+
+import "time"
+
+// OneSec repeats an operation for one second and waits 25ms in between.
```

Comment:
I didn't mind having the generic default ones with the generic names (in case we wanted to globally ratchet things down), though this is useful in reading the tests and knowing what to expect.